### PR TITLE
Switch to webpack

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract

Moves to webpack. This deletes all of the external libraries to the repo (this makes it way easier to audit the code as well as updating libraries).
Convertes all of the existing code to ES6 modules. exported things in `index.js` will be available under the MPW namespace. This is only to access them in HTML, the namespace shouldn't be used in regular javascript.
 
To install dependencies:
`npm i`. To run webpack: `npm run build`. Then, the dist folder should be hosted/opened locally.
In a future PR the global.js file should be split in smaller files.

## What features or improvements were added?
- Removed a couple of redundant libraries
- Translated the entire codebase to ES6 modules
- Integrated webpack to the codebase

## How does this benefit users?
- Performance improvements by reducing code size by dropping a couple of redundant libraries and using the webpack in-built minifier.
- Easier security audits by splitting the code into modules and dropping external minified libraries.
- Improved security by making it easier to update libraries as well as implement unit tests.

### CODE REVIEW

**Comment by JSKitty** [comment]
**Awesome!** I'm sure this is gonna be a hell of conflicts and maintenance until merged, but this is absolutely the way MPW should evolve towards. :+1:

**Review by Liquid369 — COMMENTED** [review]
tACK

Depends installs fine, the build is running fine.
Tested basic transactions sending/stake/unstake

Style nits on spacing as we're getting ready to have linting run by GitHub etc but otherwise great job!

**Review by Liquid369 — DISMISSED** [review]

File: scripts/global.js [hunk 1]
```diff
@@ -0,0 +1,933 @@
...
+    // Payment processor redirect
+    if (requestTo && requestAmount) {
+        guiPreparePayment(requestTo, requestAmount, urlParams.has('desc') ? urlParams.get('desc') : "");
+    }
+
+    // If allowed by settings: submit a simple 'hit' (app load) to Labs Analytics
+    submitAnalytics('hit');
+    setInterval(refreshChainData, 15000);
+   doms.domPrefix.value = ""
+   doms.domPrefixNetwork.innerText = cChainParams.current.PUBKEY_PREFIX.join(' or ');
```
[the remark this answers is not in the document]
Comment (Duddino) [reply scripts/global.js:194]: These spacing problems are caused by a dumb sed script I wrote

File: scripts/global.js [hunk 2]
```diff
@@ -0,0 +1,1200 @@
...
+        domTranslationSelect: document.getElementById('translation'),
+    };
+    loadImages();
+    let localTranslation = localStorage.getItem('translation');
+    // Check if set in local storage
+    if (localTranslation != null) {
+        switchTranslation(localTranslation);
+    } else {
+        // Check if we support the user's browser locale
+        if (arrActiveLangs.includes(strLang)) {
```
Comment (Liquid369) [inline scripts/global.js:171]: arrActiveLangs and strLang are being considered undefined in visual studio but it is defined in index.template.html

File: scripts/masternode.js [hunk 3]
```diff
@@ -256,21 +290,22 @@ class Masternode {
...
+            ...Masternode._numToBytes(voteCode, 4, true),
+            ...Masternode._numToBytes(sigTime, 8, true),
+        ];
+
+        const [signature, v] = await nobleSecp256k1.sign(
+            dSHA256(msg),
+            parseWIF(this.mnPrivateKey, true),
+            { der: false, recovered: true }
+        );
+        return Buffer.from([v + 27, ...signature]).toString('base64');
```
Comment (Liquid369) [inline scripts/masternode.js:308]: Buffer is undefined here and needs the import

**Review by Liquid369 — DISMISSED** [review]
A looooot better, found a few things throwing some errors in visual studio, but otherwise ACK

File: scripts/settings.js [hunk 4]
```diff
@@ -1,277 +1,333 @@
 'use strict';
 
+import {
+    doms,
+    getBalance,
+    getStakingBalance,
+    updateStakingRewardsGUI,
+    translation,
+} from './global.js';
```
Comment (Liquid369) [inline scripts/settings.js:9]: Here it is imported in global.js but in global.js it is importing from i18n.js

File: scripts/mempool.js [hunk 5]
```diff
@@ -96,27 +99,32 @@ class Mempool {
...
      * Check if an exact UTXO match can be found in our wallet
      * @param {id,path,vout} - txid path and vout of the UTXO
      * @returns {Boolean} `true` or `false`
      */
-         isAlreadyStored({id,path,vout,status}) {
-            for (const cUTXO of this.UTXOs) {
-                if (cUTXO.id === id && cUTXO.path===path && cUTXO.vout===vout && cUTXO.status===status) {
-                    return true;
-                }
+    isAlreadyStored({ id, path, vout, status }) {
```
Comment (Liquid369) [inline scripts/mempool.js:107]: With the es-lint rules this overwrites the prior declared isAlreadyStored(cNewUTXO)

We will have to rename the function and appropriate calls.

File: scripts/global.js [hunk 6]
```diff
@@ -0,0 +1,1181 @@
...
+
+        // GUI Updater
+        doms.domVanityUiButtonTxt.innerText =
+            'Stop (Searched ' + attempts.toLocaleString('en-GB') + ' keys)';
+        vanUiUpdater = setInterval(() => {
+            doms.domVanityUiButtonTxt.innerText =
+                'Stop (Searched ' + attempts.toLocaleString('en-GB') + ' keys)';
+        }, 200);
+
+        function checkResult(data) {
```
Comment (Liquid369) [inline scripts/global.js:814]: Same here with es-lint added no inner function declarations can be made, we need to fix this portion to its own function and being called here

**Review by Liquid369 — COMMENTED** [review]
And lastly the webpack.config.js is being ready as mixed with tabs and spaces

File: scripts/mempool.js (hunk 5 above)
Comment (Duddino) [reply scripts/mempool.js:107]: Yes, this will be fixed in #65

**Review by panleone — APPROVED** [review]
tACK: tested with basically every single existing button of MPW (except for ledger) , awesome PR this will finally let us to start with shield transactions.

**Review by JSKitty — APPROVED** [review]
**tACK!** :tada:

:heavy_check_mark: - Wallet Creation, Export and Import (Seed and Legacy).
:heavy_check_mark: - Send, Receive, Delegate, Undelegate.
:heavy_check_mark: - Cold Rewards recognised (Mainnet Tested Only).
:heavy_check_mark: - Masternode creation & start.
:heavy_check_mark: - All Settings.
:heavy_check_mark: - Mainnet and Testnet.

**Comment by Liquid369** [comment]
tACK. 
Screenshot 2023-01-11 at 1 57 29 PM
Screenshot 2023-01-11 at 1 57 47 PM

**Review by Liquid369 — APPROVED** [review]

**Review by panleone — APPROVED** [review]
tACK: tested again and it works fine, really good work :)

**Pull request merged**